### ISSUE TITLE
Question about the mutation process.

### ISSUE THREAD

**Issue by vladiulianbogdan** [abouchakra/Collective-Risk-Dilemma#1]
Hello,

I have some questions about the mutation process. I've looked over the C++ code and from I've seen there is only one individual mutated per generation.

```
void mutatedGeneration(std::vector<Individual> &popN,std::vector<double> & mutationP, int const& inD,int const& roundN,double * sigma,std::mt19937_64 & rng, std::uniform_real_distribution<double> & uniR,  std::uniform_int_distribution<int> & uni2)
{
    int sig=1;
    int maxG=3;
    
    if(roundN==1)maxG=1;
    std::vector<int> myDRvector;
    for (int i=0; i<inD; ++i)
    {
        myDRvector.push_back(i); // 1 2 3 4 5 6 7 8 9 vector of indiviual numbers
       
    }
    std::random_shuffle (myDRvector.begin(), myDRvector.end()); //this shuffles the individuals
    
    int rndIn=0;
    
    for(int i=0; i<maxG; i++)
    {
        int muI=mutateIndividual(mutationP, uniR(rng));
        
        for(int n=0; n<muI;n++)
        {
            int player=myDRvector[rndIn];
	    if(i==1) {sig=0;}
            else{sig=1;}

	    popN[player].Mutator(i,uni2(rng),rng, sigma[sig]);
        }
    }
	return;
}
```

The `rndIn` variable will always be 0 so the individual that will be on the first index will be mutated multiple times. Is this intentional? Also what is the purpose of the second loop         `for(int n=0; n<muI;n++)`?

**Comment by abouchakra** [comment]
Yes for exploring Probability Functions and Rich and Poor we set the parameters mutationP so that we had at most one individual chosen. We were more interested that mutation occurred at different positions (using the .Mutator function) in the strategy vector (threshold, and amounts) for that individual. 

if you put n instead of rndIn and increase mutationP then you should also have more individuals that are mutated

**Comment by vladiulianbogdan** [comment]
Thank you for your response!

I have one more question regarding the paper about Figure. 5 b).

<img width="172" alt="Screenshot 2020-01-02 at 7 46 27 PM" src="https://user-images.githubusercontent.com/2605114/71685766-90c07680-2d99-11ea-8fc3-e47a46341aa2.png">

I've interpreted that the shapes from (1) are using the lambda values at (2) in the order that they appear. However, the shapes should be in the opposite order if that is the case. For example, the second shape from the list of shapes corresponds to the lambda = 10 and the forth one is for lambda = 0.1. Did I understand it correctly?
